Unexpected NPE in dotty.tools.scaladoc.translators.SignatureBuilder

## Compiler version

Scala version: 3.7.3
Java: 24.0.1

## Minimized code

Not sure how to minimize the code. I started running into the issue when I did `sbt-native-packager` setup and tried to `Docker / stage` project, it suddently start failing on `Compile / doc` step (not event sure why docker stage needs scaladoc). Because I never even tried to compile doc before for the project, I cannot locate if this related to my code or 3rd party lib. My code doesn't have any doc comments though. One thing for sure is it's not expected behavior for scaladoc translator. Maybe any advice where to look for the culprit? 

## Output

```
[error] java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
[error]         at xsbt.DottydocRunner.run(DottydocRunner.java:65)
[error]         at xsbt.ScaladocInterface.run(ScaladocInterface.java:11)
[error]         at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
[error]         at java.base/java.lang.reflect.Method.invoke(Method.java:565)
[error]         at sbt.internal.inc.AnalyzingCompiler.invoke(AnalyzingCompiler.scala:329)
[error]         at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:175)
[error]         at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:133)
[error]         at sbt.Doc$.$anonfun$scaladoc$1(Doc.scala:53)
[error]         at sbt.Doc$.$anonfun$scaladoc$1$adapted(Doc.scala:41)
[error]         at sbt.RawCompileLike$.$anonfun$prepare$1(RawCompileLike.scala:80)
[error]         at sbt.RawCompileLike$.$anonfun$prepare$1$adapted(RawCompileLike.scala:73)
[error]         at sbt.RawCompileLike$.$anonfun$cached$4(RawCompileLike.scala:64)
[error]         at sbt.RawCompileLike$.$anonfun$cached$4$adapted(RawCompileLike.scala:62)
[error]         at sbt.util.Tracked$.$anonfun$inputChangedW$1(Tracked.scala:220)
[error]         at sbt.RawCompileLike$.$anonfun$cached$1(RawCompileLike.scala:69)
[error]         at sbt.RawCompileLike$.$anonfun$cached$1$adapted(RawCompileLike.scala:53)
[error]         at sbt.Defaults$.$anonfun$docTaskSettings$4(Defaults.scala:2277)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:63)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:69)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:283)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:24)
[error]         at sbt.Execute.work(Execute.scala:292)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:283)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:65)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
[error]         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:545)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1095)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:619)
[error]         at java.base/java.lang.Thread.run(Thread.java:1447)
[error] Caused by: java.lang.reflect.InvocationTargetException
[error]         at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:116)
[error]         at java.base/java.lang.reflect.Method.invoke(Method.java:565)
[error]         at xsbt.DottydocRunner.run(DottydocRunner.java:61)
[error]         at xsbt.ScaladocInterface.run(ScaladocInterface.java:11)
[error]         at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
[error]         at java.base/java.lang.reflect.Method.invoke(Method.java:565)
[error]         at sbt.internal.inc.AnalyzingCompiler.invoke(AnalyzingCompiler.scala:329)
[error]         at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:175)
[error]         at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:133)
[error]         at sbt.Doc$.$anonfun$scaladoc$1(Doc.scala:53)
[error]         at sbt.Doc$.$anonfun$scaladoc$1$adapted(Doc.scala:41)
[error]         at sbt.RawCompileLike$.$anonfun$prepare$1(RawCompileLike.scala:80)
[error]         at sbt.RawCompileLike$.$anonfun$prepare$1$adapted(RawCompileLike.scala:73)
[error]         at sbt.RawCompileLike$.$anonfun$cached$4(RawCompileLike.scala:64)
[error]         at sbt.RawCompileLike$.$anonfun$cached$4$adapted(RawCompileLike.scala:62)
[error]         at sbt.util.Tracked$.$anonfun$inputChangedW$1(Tracked.scala:220)
[error]         at sbt.RawCompileLike$.$anonfun$cached$1(RawCompileLike.scala:69)
[error]         at sbt.RawCompileLike$.$anonfun$cached$1$adapted(RawCompileLike.scala:53)
[error]         at sbt.Defaults$.$anonfun$docTaskSettings$4(Defaults.scala:2277)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:63)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:69)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:283)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:24)
[error]         at sbt.Execute.work(Execute.scala:292)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:283)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:65)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
[error]         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:545)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1095)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:619)
[error]         at java.base/java.lang.Thread.run(Thread.java:1447)
[error] Caused by: java.lang.NullPointerException: Cannot invoke "scala.collection.immutable.List.$plus$plus(scala.collection.IterableOnce)" because the return value of "dotty.tools.scaladoc.translators.SignatureBuilder.content()" is null
[error]         at dotty.tools.scaladoc.translators.SignatureBuilder.signature(ScalaSignatureUtils.scala:10)
[error]         at dotty.tools.scaladoc.translators.SignatureBuilder.termParamList$$anonfun$1(ScalaSignatureUtils.scala:89)
[error]         at dotty.tools.scaladoc.translators.SignatureBuilder.list(ScalaSignatureUtils.scala:23)
[error]         at dotty.tools.scaladoc.translators.SignatureBuilder.termParamList(ScalaSignatureUtils.scala:85)
[error]         at dotty.tools.scaladoc.translators.SignatureBuilder.functionParameters$$anonfun$1(ScalaSignatureUtils.scala:80)
[error]         at dotty.tools.scaladoc.translators.SignatureBuilder.list$$anonfun$1(ScalaSignatureUtils.scala:23)
[error]         at scala.collection.LinearSeqOps.foldLeft(LinearSeq.scala:183)
[error]         at scala.collection.LinearSeqOps.foldLeft$(LinearSeq.scala:179)
[error]         at scala.collection.immutable.List.foldLeft(List.scala:79)
[error]         at dotty.tools.scaladoc.translators.SignatureBuilder.list(ScalaSignatureUtils.scala:23)
[error]         at dotty.tools.scaladoc.translators.SignatureBuilder.functionParameters(ScalaSignatureUtils.scala:79)
[error]         at dotty.tools.scaladoc.translators.ScalaSignatureProvider.methodLikeSignature(ScalaSignatureProvider.scala:109)
[error]         at dotty.tools.scaladoc.translators.ScalaSignatureProvider.methodSignature(ScalaSignatureProvider.scala:118)
[error]         at dotty.tools.scaladoc.translators.ScalaSignatureProvider.rawSignature(ScalaSignatureProvider.scala:16)
[error]         at dotty.tools.scaladoc.renderers.MemberRenderer.memberSignature(MemberRenderer.scala:180)
[error]         at dotty.tools.scaladoc.renderers.MemberRenderer.member(MemberRenderer.scala:260)
[error]         at dotty.tools.scaladoc.renderers.MemberRenderer.$anonfun$23(MemberRenderer.scala:290)
[error]         at scala.collection.immutable.List.map(List.scala:247)
[error]         at scala.collection.immutable.List.map(List.scala:79)
[error]         at dotty.tools.scaladoc.renderers.MemberRenderer.actualGroup(MemberRenderer.scala:288)
[error]         at dotty.tools.scaladoc.renderers.MemberRenderer.$anonfun$24(MemberRenderer.scala:335)
[error]         at scala.collection.immutable.List.map(List.scala:251)
[error]         at scala.collection.immutable.List.map(List.scala:79)
[error]         at dotty.tools.scaladoc.renderers.MemberRenderer.buildGroup(MemberRenderer.scala:324)
[error]         at dotty.tools.scaladoc.renderers.MemberRenderer.buildMembers(MemberRenderer.scala:438)
[error]         at dotty.tools.scaladoc.renderers.MemberRenderer.fullMember(MemberRenderer.scala:564)
[error]         at dotty.tools.scaladoc.renderers.Renderer.renderContent(Renderer.scala:136)
[error]         at dotty.tools.scaladoc.renderers.HtmlRenderer.pageContent(HtmlRenderer.scala:13)
[error]         at dotty.tools.scaladoc.renderers.Renderer.renderPage(Renderer.scala:166)
[error]         at dotty.tools.scaladoc.renderers.Renderer.renderPage$$anonfun$1(Renderer.scala:167)
[error]         at scala.collection.immutable.List.flatMap(List.scala:294)
[error]         at scala.collection.immutable.List.flatMap(List.scala:79)
[error]         at dotty.tools.scaladoc.renderers.Renderer.renderPage(Renderer.scala:167)
[error]         at dotty.tools.scaladoc.renderers.Renderer.renderPage$$anonfun$1(Renderer.scala:167)
[error]         at scala.collection.immutable.List.flatMap(List.scala:294)
[error]         at scala.collection.immutable.List.flatMap(List.scala:79)
[error]         at dotty.tools.scaladoc.renderers.Renderer.renderPage(Renderer.scala:167)
[error]         at dotty.tools.scaladoc.renderers.Renderer.renderPage$$anonfun$1(Renderer.scala:167)
[error]         at scala.collection.immutable.List.flatMap(List.scala:294)
[error]         at scala.collection.immutable.List.flatMap(List.scala:79)
[error]         at dotty.tools.scaladoc.renderers.Renderer.renderPage(Renderer.scala:167)
[error]         at dotty.tools.scaladoc.renderers.Renderer.renderPage$$anonfun$1(Renderer.scala:167)
[error]         at scala.collection.immutable.List.flatMap(List.scala:294)
[error]         at scala.collection.immutable.List.flatMap(List.scala:79)
[error]         at dotty.tools.scaladoc.renderers.Renderer.renderPage(Renderer.scala:167)
[error]         at dotty.tools.scaladoc.renderers.Renderer.$anonfun$8(Renderer.scala:154)
[error]         at scala.collection.immutable.List.map(List.scala:247)
[error]         at scala.collection.immutable.List.map(List.scala:79)
[error]         at dotty.tools.scaladoc.renderers.Renderer.render(Renderer.scala:154)
[error]         at dotty.tools.scaladoc.renderers.HtmlRenderer.render(HtmlRenderer.scala:52)
[error]         at dotty.tools.scaladoc.Scaladoc$.run(Scaladoc.scala:241)
[error]         at dotty.tools.scaladoc.Scaladoc$.run$$anonfun$1(Scaladoc.scala:70)
[error]         at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error]         at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error]         at scala.Option.map(Option.scala:242)
[error]         at dotty.tools.scaladoc.Scaladoc$.run(Scaladoc.scala:55)
[error]         at dotty.tools.dottydoc.Main$.process(Main.scala:25)
[error]         at dotty.tools.dottydoc.Main.process(Main.scala)
[error]         at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
[error]         at java.base/java.lang.reflect.Method.invoke(Method.java:565)
[error]         at xsbt.DottydocRunner.run(DottydocRunner.java:61)
[error]         at xsbt.ScaladocInterface.run(ScaladocInterface.java:11)
[error]         at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
[error]         at java.base/java.lang.reflect.Method.invoke(Method.java:565)
[error]         at sbt.internal.inc.AnalyzingCompiler.invoke(AnalyzingCompiler.scala:329)
[error]         at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:175)
[error]         at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:133)
[error]         at sbt.Doc$.$anonfun$scaladoc$1(Doc.scala:53)
[error]         at sbt.Doc$.$anonfun$scaladoc$1$adapted(Doc.scala:41)
[error]         at sbt.RawCompileLike$.$anonfun$prepare$1(RawCompileLike.scala:80)
[error]         at sbt.RawCompileLike$.$anonfun$prepare$1$adapted(RawCompileLike.scala:73)
[error]         at sbt.RawCompileLike$.$anonfun$cached$4(RawCompileLike.scala:64)
[error]         at sbt.RawCompileLike$.$anonfun$cached$4$adapted(RawCompileLike.scala:62)
[error]         at sbt.util.Tracked$.$anonfun$inputChangedW$1(Tracked.scala:220)
[error]         at sbt.RawCompileLike$.$anonfun$cached$1(RawCompileLike.scala:69)
[error]         at sbt.RawCompileLike$.$anonfun$cached$1$adapted(RawCompileLike.scala:53)
[error]         at sbt.Defaults$.$anonfun$docTaskSettings$4(Defaults.scala:2277)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:63)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:69)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:283)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:24)
[error]         at sbt.Execute.work(Execute.scala:292)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:283)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:65)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
[error]         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:545)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1095)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:619)
[error]         at java.base/java.lang.Thread.run(Thread.java:1447)
```


## Expectation

No exception

--- a/scaladoc/src/dotty/tools/scaladoc/translators/ScalaSignatureUtils.scala
+++ b/scaladoc/src/dotty/tools/scaladoc/translators/ScalaSignatureUtils.scala
@@ -1,7 +1,10 @@
 package dotty.tools.scaladoc
 package translators
 
+import dotty.tools.dotc.util.chaining.*
+
 case class SignatureBuilder(content: Signature = Nil) extends ScalaSignatureUtils:
+  assert(content != null)
   def plain(str: String): SignatureBuilder = copy(content = content :+ Plain(str))
   def name(str: String, dri: DRI, isCaptureVar: Boolean = false/*under CC*/): SignatureBuilder =
     val suffix = if isCaptureVar then List(Keyword("^")) else Nil
@@ -24,7 +27,10 @@ case class SignatureBuilder(content: Signature = Nil) extends ScalaSignatureUtil
   ): SignatureBuilder = elements match {
     case Nil => if forcePrefixAndSuffix then signature(prefix).signature(suffix) else this
     case head :: tail =>
-      tail.foldLeft(elemOp(signature(prefix), head))((b, e) => elemOp(b.signature(separator), e)).signature(suffix)
+      tail.foldLeft(elemOp(signature(prefix), head)): (b, e) =>
+        val sepped = b.signature(separator)
+        elemOp(sepped, e)
+      .signature(suffix)
   }
 
   def annotationsBlock(d: Member): SignatureBuilder =
@@ -92,9 +98,13 @@ case class SignatureBuilder(content: Signature = Nil) extends ScalaSignatureUtil
 
   def termParamList(params: TermParameterList) =
     this.list(params.parameters, prefix = List(Plain("("), Keyword(params.modifiers)), suffix = List(Plain(")")), forcePrefixAndSuffix = true) { (bld, p) =>
-      val annotationsAndModifiers = bld.annotationsInline(p)
-        .keyword(p.modifiers)
-      val name = p.name.fold(annotationsAndModifiers)(annotationsAndModifiers.name(_, p.dri).plain(": "))
+      val annotationsAndModifiers =
+        bld.annotationsInline(p)
+          .keyword(p.modifiers)
+      val name = p.name match {
+        case Some(name) => annotationsAndModifiers.name(name, p.dri).plain(": ")
+        case none => annotationsAndModifiers
+      }
       name.signature(p.signature)
     }
 

